tests/unit/verifiers/test_ovn_central.py

```diff
@@ -0,0 +1,783 @@
+
+    # Determine expected leaders
+    expected_sb_leader = expected_nb_leader = None
+    for status_pair in sample_cluster_data.values():
+        if expected_nb_leader and expected_sb_leader:
+            break
+        for cluster, status in status_pair.items():
+            if status.is_leader and cluster == "southbound":
+                expected_sb_leader = status.short_id
+            if status.is_leader and cluster == "northbound":
+                expected_nb_leader = status.short_id
+
+    # prepare expected positive result
+    pass_msg = "All units agree that {} is {} leader."
+    pass_result = Result()
+    pass_result.add_partial_result(
+        Severity.OK, pass_msg.format(expected_sb_leader, "Southbound")
+    )
+    pass_result.add_partial_result(
+        Severity.OK, pass_msg.format(expected_nb_leader, "Northbound")
+    )
+
+    verify_leader_consistency(pass_result, sample_cluster_data, mocker, model)
+
+
+def test_ovn_central_check_leader_consistency_disagreement(model, mocker):
+    """Test check verifies that error is raised if some units disagree on leader.
+
+    Every unit must agree on a leader in both Southbound and Northbound cluster,
+    otherwise a FAIL result is returned by "check_leader_consistency".
+    """
+    unit_names = [unit for unit in model.units if unit.startswith("ovn-central")]
+
+    #  Generated sample data must be modified to create leader discrepancy
+    sample_cluster_data = generate_cluster_status(unit_names, "ClusterStatus")
+    leaders = unit_names[0:2]
+    sb_support_map = defaultdict(list)
+    nb_support_map = defaultdict(list)
+    for unit, status_pair in sample_cluster_data.items():
+        for cluster, status in status_pair.items():
+            if unit in leaders:
+                status.leader = "self"
+                status.vote = "self"
+            leader = status.short_id if status.is_leader else status.leader
+            if cluster == "southbound":
+                sb_support_map[leader].append(unit)
+            elif cluster == "northbound":
+                nb_support_map[leader].append(unit)
+
+    # prepare expected negative result
+    sb_fail_msg = "There's no consensus on Southbound cluster leader. "
+    for leader, units in sb_support_map.items():
+        unit_list = ", ".join(units)
+        sb_fail_msg += f"{leader} is supported by {unit_list}; "
+
+    nb_fail_msg = "There's no consensus on Northbound cluster leader. "
+    for leader, units in nb_support_map.items():
+        unit_list = ", ".join(units)
+        nb_fail_msg += f"{leader} is supported by {unit_list}; "
+
+    fail_result = Result()
+    fail_result.add_partial_result(Severity.FAIL, sb_fail_msg)
+    fail_result.add_partial_result(Severity.FAIL, nb_fail_msg)
+
+    verify_leader_consistency(fail_result, sample_cluster_data, mocker, model)
+
+
+def test_ovn_central_check_leader_consistency_no_leader(model, mocker):
```

Comment:
honestly, I am a little lost in the code how to generate different cluster data. So I didn't check that part in detail.
But it seems it cover these test cases

- consistency leader
- inconsistency leader
- no leader

It's good to me.

Reply:
Yeah, I tried to make this as a single parametrize test but it only made the test more convoluted. I separated it into three different tests because manipulating the cluster status to create inconsistency and to create "no leader" situations are quite different.